managed-ledger/src/main/java/org/apache/bookkeeper/mledger/impl/ManagedCursorImpl.java

```diff
@@ -2156,47 +2156,55 @@ void readOperationCompleted() {
     }
 
     void asyncDeleteLedger(final LedgerHandle lh) {
-        if (lh == null) {
+        asyncDeleteLedger(lh, 3);
+    }
+    
+    private void asyncDeleteLedger(final LedgerHandle lh, int retry) {
+        if (lh == null || retry <= 0) {
```

Comment:
If retry <=0 do you want to print a warning log saying unable to delete ledger programattically - need to delete manually.

Reply:
sure. I will add the log.